main.go

```diff
@@ -209,20 +153,39 @@ func mainWithExitCode() exitCode {
 		return failure("Failed to parse the included tracers: %v", err)
 	}
 
-	log.Infof("Assigned ProjectID: %d HostID: %d", args.projectID, environment.HostID())
+	metadataCollector := hostmetadata.NewCollector(args.collAgentAddr)
+	metadataCollector.AddCustomData("os.type", "linux")
 
-	metadataCollector := hostmetadata.NewCollector(args.collAgentAddr, environment)
+	kernelVersion, err := getKernelVersion()
+	if err != nil {
+		return failure("Failed to get Linux kernel version: %v", err)
+	}
+	// OTel semantic introduced in https://github.com/open-telemetry/semantic-conventions/issues/66
+	metadataCollector.AddCustomData("os.kernel.release", kernelVersion)
 
-	// TODO: Maybe abort execution if (some) metadata can not be collected
-	hostMetadataMap = metadataCollector.GetHostMetadata()
+	// hostname and sourceIP will be populated from the root namespace.
+	var hostname, sourceIP string
+
+	if err = runInRootNS(func() error {
+		var hostnameErr error
+		hostname, hostnameErr = os.Hostname()
+		if hostnameErr != nil {
+			return fmt.Errorf("failed to get hostname: %v", hostnameErr)
+		}
 
-	if bpfJITEnabled, found := hostMetadataMap["host.sysctl.net.core.bpf_jit_enable"]; found {
-		if bpfJITEnabled == "0" {
-			log.Warnf("The BPF JIT is disabled (net.core.bpf_jit_enable = 0). " +
-				"Enable it to reduce CPU overhead.")
+		srcIP, ipErr := getSourceIPAddress(args.collAgentAddr)
+		if ipErr != nil {
+			return fmt.Errorf("failed to get source IP: %v", ipErr)
 		}
+		sourceIP = srcIP.String()
+		return nil
+	}); err != nil {
+		return failure("Failed to get information in root namespace: %v", err)
```

Comment:
Before the recent hostmetadata changes, we didn't abort agent startup if we couldn't retrieve `sourceIP` / `hostname`. With the hostmetadata changes we introduced a bug: we kept the old behavior but assumed that `hostname` and `sourceIP` are always successfully retrieved which could crash the agent on startup (the pre-hostmetadata-changes logic always initialized these to the empty string but was removed). 

This PR removes the potential crash by returning failure but since these values are only
used as metadata and are not hard requirements, we should probably continue here on failures and restore the pre-hostmetadata-change behavior: initialize to the empty string if they can't retrieved (since `reporter.Start` expects them). We could do that in a new PR.

Reply:
Downgraded `failure()` to `log.Warnf()` with https://github.com/elastic/otel-profiling-agent/pull/90/commits/3b25312f79deb27fbecebf00d95b55d1924f6a21.